### PR TITLE
Cleanup readmes

### PR DESCRIPTION
**Pull request by dcmiddle**
Consolidated and updated readme files and project documentation.

Reorganized the project documentation (community stuff for example; not the API docs) to facilitate future publication to github pages or readthedocs or etc.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,5 +1,4 @@
-<p align="center"><h1>Hyperledger URSA</h1></p><br>
----
+<p align="center"><h1>Hyperledger URSA</h1></p><br>
```
Comment (danintel) [inline README.md]: It seems weird to have HTML in a MD file--I suggest just `# Hyperledger URSA` 
The centering is ignored anyway.

File: README.md [hunk 2]
```diff
@@ -11,46 +10,69 @@
...
+the worry of compatibility issues from C code. For this reason, Ursa can be
+compiled with *portable* mode which replaces any external libraries with rust
+compatible code. Ursa developers take care when choosing suitable replacements
+that are cryptographically safe to use but may not have been audited and vetted
+in a similar manner to these external libraries. Ursa consumers should note this
+when using portable mode for their applications.
 
 # Building from Source
-Libursa and Libzmix rely on libsodium for the default secure mode. Please see the following document for specific
-platform installations [here](docs/build-enviroment.md).
```
Comment (danintel) [inline README.md]: Lets spell our filenames correctly. Move `docs/build-enviroment.md` to `docs/build-environment.md` and link to the correctly-spelled name.

File: README.md [hunk 3]
```diff
@@ -11,46 +10,69 @@
...
+
+Libursa and Libzmix rely on libsodium for the default
+secure mode. Please see the following document for specific platform
+installations [here](docs/build-enviroment.md).
```
Comment (danintel) [inline README.md]: Same as above.

File: docs/source/community/join_the_discussion.rst [hunk 4]
```diff
@@ -10,7 +10,7 @@ help to involved discussions. Be mindful that chat can sometimes disclude partic
 available at that time. Discussions with broad impact to the community should take place on the
 mail list. 
 
-  `#sawtooth <https://chat.hyperledger.org/channel/crypto-lib>`_
```
Comment (danintel) [inline docs/source/community/join_the_discussion.rst]: Line 9 above: s/disclude/exclude/

File: docs/source/community/join_the_discussion.rst [hunk 5]
```diff
@@ -20,7 +20,7 @@ Mailing Lists
 The Hyperledger Ursa mailing list is the prefered place for discussions about decisional matters
```
Comment (danintel) [inline docs/source/community/join_the_discussion.rst]: s/prefered/preferred/

File: README.md (hunk 1 above)
Comment (dcmiddle) [reply README.md]: Fixed

File: README.md (hunk 2 above)
Comment (dcmiddle) [reply README.md]: Fixed

File: README.md (hunk 3 above)
Comment (dcmiddle) [reply README.md]: Fixed

File: docs/source/community/join_the_discussion.rst (hunk 4 above)
Comment (dcmiddle) [reply docs/source/community/join_the_discussion.rst]: Fixed

File: docs/source/community/join_the_discussion.rst (hunk 5 above)
Comment (dcmiddle) [reply docs/source/community/join_the_discussion.rst]: Fixed

**Comment by dcmiddle** [comment]
rebased over master

**Pull request merged**